TESReloaded/Core/ShadowManager.cpp

```diff
@@ -14,11 +17,22 @@ void ShadowManager::Initialize() {
 	TheShadowManager->ShadowMapPixel = (ShaderRecordPixel*)ShaderRecord::LoadShader("ShadowMap.pso", NULL);
 	TheShadowManager->ShadowCubeMapVertex = (ShaderRecordVertex*)ShaderRecord::LoadShader("ShadowCubeMap.vso", NULL);
 	TheShadowManager->ShadowCubeMapPixel = (ShaderRecordPixel*)ShaderRecord::LoadShader("ShadowCubeMap.pso", NULL);
-    
-    ShadowMapSize = ShadowsExteriors->ShadowMapSize[0];
+
+	// Store Shadow map sizes in Constants to pass to the Shaders
+
+	TheShaderManager->ShaderConst.ShadowMap.ShadowMapRadius.x = ShadowsExteriors->ShadowMapRadius[0];
+	TheShaderManager->ShaderConst.ShadowMap.ShadowMapRadius.y = ShadowsExteriors->ShadowMapRadius[1];
+	TheShaderManager->ShaderConst.ShadowMap.ShadowMapRadius.z = ShadowsExteriors->ShadowMapRadius[2];
+	TheShaderManager->ShaderConst.ShadowMap.ShadowMapRadius.w = ShadowsExteriors->ShadowMapRadius[3];
+
+	ShadowMapSize = ShadowsExteriors->ShadowMapSize[0];
     TheShaderManager->CreateFrameVertex(ShadowMapSize, ShadowMapSize, &TheShadowManager->BlurShadowVertex[0]);
-    ShadowMapSize = ShadowsExteriors->ShadowMapSize[1];
-    TheShaderManager->CreateFrameVertex(ShadowMapSize, ShadowMapSize, &TheShadowManager->BlurShadowVertex[1]);
+	ShadowMapSize = ShadowsExteriors->ShadowMapSize[1];
+	TheShaderManager->CreateFrameVertex(ShadowMapSize, ShadowMapSize, &TheShadowManager->BlurShadowVertex[1]);
+	ShadowMapSize = ShadowsExteriors->ShadowMapSize[2];
+	TheShaderManager->CreateFrameVertex(ShadowMapSize, ShadowMapSize, &TheShadowManager->BlurShadowVertex[2]);
+	ShadowMapSize = ShadowsExteriors->ShadowMapSize[3];

```

Comment:
you can create these FrameVertex in the loop

Reply:
Like this? 

(line 36)
```
	for (int i = 0; i < 5; i++) {
		ShadowMapSize = ShadowsExteriors->ShadowMapSize[i];
		TheShadowManager->ShadowMapViewPort[i] = { 0, 0, ShadowMapSize, ShadowMapSize, 0.0f, 1.0f };
        TheShadowManager->ShadowMapInverseResolution[i] = 1.0f / (float) ShadowMapSize;
		TheShaderManager->CreateFrameVertex(ShadowMapSize, ShadowMapSize, &TheShadowManager->BlurShadowVertex[i]);
	}
```